pkg/cfn/builder/managed_nodegroup_test.go

```diff
@@ -94,6 +94,55 @@ func TestManagedResources(t *testing.T) {
 
 }
 
+func TestManagedNodeRole(t *testing.T) {
+	nodeRoleTests := []struct {
+		description      string
+		nodeGroup        *api.ManagedNodeGroup
+		expectedNodeRole string
+	}{
+		{
+			description: "InstanceRoleARN is not provided",
+			nodeGroup: &api.ManagedNodeGroup{
+				ScalingConfig: &api.ScalingConfig{},
+				SSH: &api.NodeGroupSSH{
+					Allow: api.Disabled(),
+				},
+				IAM: &api.NodeGroupIAM{
+				},
+			},
+			expectedNodeRole: "NodeInstanceRole", // creating new role
+		},
+		{
+			description: "InstanceRoleARN is provided",
+			nodeGroup: &api.ManagedNodeGroup{
+				ScalingConfig: &api.ScalingConfig{},
+				SSH: &api.NodeGroupSSH{
+					Allow: api.Disabled(),
+				},
+				IAM: &api.NodeGroupIAM{
+					InstanceRoleARN: "arn::DUMMY::DUMMYROLE",
+				},
+			},
+			expectedNodeRole: "arn::DUMMY::DUMMYROLE", // using the provided role
+		},
+	}
+
+	for i, tt := range nodeRoleTests {
+		t.Run(fmt.Sprintf("%d: %s", i, tt.description), func(t *testing.T) {
+			stack := NewManagedNodeGroup(api.NewClusterConfig(), tt.nodeGroup, "iam-test")
+			err := stack.AddAllResources()
+			assert.NoError(t, err)
+
+			bytes, err := stack.RenderJSON()
+			assert.NoError(t, err)
+
+			_, err = goformation.ParseJSON(bytes)
+			assert.NoError(t, err)
+			assert.Contains(t, string(bytes), tt.expectedNodeRole)
```

Comment:
A `contains` check on the CloudFormation template can result in false-positives if, e.g., `ManagedNodeGroup.Name` is set to `NodeInstanceRole`. We can be more specific here by checking the existence of a `NodeInstanceRole` resource, or lack thereof, or asserting `ManagedNodeGroup.NodeRole == expectedNodeRole` when a custom instance role is supplied.

Reply:
Agreed with your comment. I was trying to parse the template and assert `NodeGroup` resource, but the custom goformation library we are using didn't support for this resource. 

The best I can do right now is to check if there is any `NodeInstanceRole` IAM role resource got created as part of CF.